contracts/test/CompoundGovernorCountingFractional.t.sol

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.26;
+
+import {ProposalTest} from "contracts/test/helpers/ProposalTest.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+import {GovernorCountingFractionalUpgradeable} from "contracts/CompoundGovernor.sol";
+
+abstract contract CountingMode is ProposalTest {
+    function testFuzz_ReturnsCorrectCountingMode() public view {
+        assertEq(governor.COUNTING_MODE(), "support=bravo,fractional&quorum=for,abstain&params=fractional");
+    }
+}
+
+abstract contract HasVoted is ProposalTest {
+    function testFuzz_ReturnsCorrectVotingStatus(
+        bool _hasVotedBefore,
+        uint256 _voteSeed,
+        uint256 _proposerIndex,
+        uint256 _voterIndex
+    ) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
+        _voterIndex = bound(_voterIndex, 0, _majorDelegates.length - 1);
+        address _proposer = _majorDelegates[_proposerIndex];
+        address _voter = _majorDelegates[_voterIndex];
+
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = _getProposalId(_proposal);
+        _submitProposal(_proposer, _proposal);
+
+        vm.prank(_voter);
+        if (_hasVotedBefore) {
```

Comment:
Nit: Because we choose to vote if the boolean variable is true, shouldn't the name of the variable be something like "hasNotVotedBefore" ?  I think that here for test execution the name doesn't matter as we're using it for fuzzing whether to vote or not, and because each fuzz run starts with fresh state, we won't double vote.. but reading the code, I think the name as it is can confuse the reader.